lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +231,53 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        link = article_bs.find('a', class_="card-big").get('href')
+        url = self.url_pattern + link[len('/articles')::]
+        return url
+
+        #link_tag = article_bs.find('a', class_='card-big')
+        #if link_tag and 'href' in link_tag.attrs:
+        #    link = link_tag['href']
+        #    url = self.url_pattern + link[len('/articles'):]
+        #    return url
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        #seed_urls = self.get_search_urls()
+        #for seed_url in seed_urls:
+        #    response = make_request(seed_url, self.config)
+#
+        #    if not response.ok:
+        #        continue
+#
+        #    article_bs = BeautifulSoup(response.text, 'html.parser')
+#
+        #    for link in article_bs.find(class_="col-md-7").find_all(class_='articles__list'):
+        #        if len(self.urls) == self.config.get_num_articles():
+        #            break
+        #        if self._extract_url(link) and self._extract_url(link) not in self.urls:
+        #            self.urls.append(self._extract_url(link))
+
+        seed_urls = self.get_search_urls()
+        for seed_url in seed_urls:
+            response = make_request(seed_url, self.config)
+            if not response.ok:
+                continue
+
+            article_bs = BeautifulSoup(response.text, 'html.parser')
+            articles = article_bs.find_all('div', class_='col-md-7')
+            for article in articles:
+                links = article.find_all('div', class_='articles__list')
+                for link in links:
+                    extracted_url = self._extract_url(link)
+                    if extracted_url and extracted_url not in self.urls:
+                        self.urls.append(extracted_url)
+                    if len(self.urls) >= self.config.get_num_articles():
+                        break
+                if len(self.urls) >= self.config.get_num_articles():
+                    break
```

Comment:
You don't break from the cycle over `seed_urls`, so even when you've reached desired number of articles, you'll continue to make requests with the next seed URL, am I right?